src/violations.rs

```diff
@@ -4253,6 +4253,20 @@ impl AlwaysAutofixableViolation for EndsInPeriod {
     }
 }
 
+define_violation!(
+    pub struct UseImperativeMood(pub String);
+);
+impl Violation for UseImperativeMood {
+    fn message(&self) -> String {
+        let UseImperativeMood(first_line) = self;
+        format!("First line of docstring should be in imperative mood: \"{first_line}\"")
+    }
+
+    fn placeholder() -> Self {
+        UseImperativeMood("Returns a thing".to_string())
+    }
+}
```

Comment:
The violations are only all defined in `violations.rs` for historical reasons. We actually want to move these next to the rule implementations in `rules::`. Also you'll need to remove the placeholder and add `#[derive_message_formats]` to the `message` function.

Reply:
Yeah I agree that updating `CONTRIBUTING.md` would make sense ... however I am currently actively figuring out a better structure for ruff rules (moving away from the upstream-centric architecture since many rules are implemented by several linters), so the recommended structure will probably change again in the near future ... at which point I'll certainly make an effort to properly document it :)

Reply:
Sure, I get that ruff is a very fast-moving target :)

It's just that that's where new contributors (yours truly) look for instructions, and getting contradicting information in a PR review (especially with a "you'll need to" (though maybe it wasn't meant that way)) is a bit unfortunate.

Reply:
Right I get what you mean :)

Now that you have an `imperative_mood.rs` file you could also move the violation definition there like I suggested.